Cargo.lock

```diff
@@ -2031,7 +2484,16 @@ version = "0.2.16"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "fb5a58c1855b4b6819d59012155603f0b22ad30cad752600aadfcb695265519a"
 dependencies = [
- "bitflags",
+ "bitflags 1.3.2",
+]
+
+[[package]]
```

Comment:
This is a package for RedoxOS, which is not even supported by the blockchain.  Why is this a dependency?

Reply:
I added `wsts` and `p256k1` as deps under `./stackslib` but I'm not sure how these got here. I will investigate...

Reply:
The `Cargo.lock` file on `master` also contains `redox_syscall` and `redox_users`. See [here](https://github.com/stacks-network/stacks-blockchain/blob/e53ba10e75dc26f3dfe92226df2862f1167f806b/Cargo.lock#L2102). Here's the full dependency chain:

```
slog-term -> term -> dirs-next -> dirs-sys-next -> redox_users -> redox_syscall
```

In this PR, whatever I added needs an *additional* version of `redox_syscall` to be specified. This adds the new dependency chain:
```
pk256k1 -> rustfmt-wrapper -> tempfile -> redox_syscall
```

IMO it doesn't matter because as far as I can tell it's never downloaded and built. I don't see it using `cargo tree` or anywhere under `./target`. I guess it's disabled by a feature flag but the dependency gets resolved and pinned anyway?

Reply:
Huh, what an odd behavior on Cargo's part.

As long as it's not getting linked into the binary, then I guess this is fine. I wish Cargo would only list packages that actually get built in the `.lock` file.